pkg/gitprovider/aws-codecommit.go

```diff
@@ -291,6 +267,74 @@ func (g *AwsCodeCommitGitProvider) getPrContext(staticContext *StaticGitContext)
 	return &repo, nil
 }
 
+func (g *AwsCodeCommitGitProvider) GetBranchByCommit(staticContext *StaticGitContext) (string, error) {
+	if staticContext.Sha == nil || *staticContext.Sha == "" {
+		return *staticContext.Branch, nil
+	}
+
+	client, err := g.getApiClient()
+	if err != nil {
+		return "", err
+	}
+
+	branches, err := client.ListBranches(context.TODO(), &codecommit.ListBranchesInput{
+		RepositoryName: aws.String(staticContext.Name),
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to list branches, %v", err)
+	}
+	var branchName string
+	for _, branch := range branches.Branches {
+		branchInfo, err := client.GetBranch(context.TODO(), &codecommit.GetBranchInput{
+			RepositoryName: aws.String(staticContext.Name),
+			BranchName:     aws.String(branch),
+		})
+		if err != nil {
+			log.Infof("failed to get branch info for %s, %v", branch, err)
+			continue
+		}
+
+		if *staticContext.Sha == *branchInfo.Branch.CommitId {
+			branchName = branch
+			break
+		}
+
+		commitID := staticContext.Sha
+		for commitID != nil {
+			commit, err := client.GetCommit(context.Background(), &codecommit.GetCommitInput{
```

Comment:
@the-johnwick again here, you have a for loop going through branches and then a for loop for commits but the commit query doesn't even include the branch. Can this be reduced?